### ISSUE TITLE
Test failure: LockedAppOfflineDroppedWhileSiteIsDown_SiteReturns503

### ISSUE THREAD

**Issue by ryanbrandenburg** [aspnet/IISIntegration#1204]
This test [fails](http://aspnetci/viewLog.html?buildId=527343&buildTypeId=XPlat_Windows_Win7_Universe) occasionally with the following error:
```
System.AggregateException : One or more errors occurred. (IIS failed to start site.) (Unable to load DLL 'api-ms-win-service-core-l1-1-1.dll' or one of its dependencies: The specified module could not be found. (Exception from HRESULT: 0x8007007E))
---- System.TimeoutException : IIS failed to start site.
---- System.DllNotFoundException : Unable to load DLL 'api-ms-win-service-core-l1-1-1.dll' or one of its dependencies: The specified module could not be found. (Exception from HRESULT: 0x8007007E)
----- Inner Stack Trace #1 (System.TimeoutException) -----
   at Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer.WaitUntilSiteStarted() in /_/src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployer.cs:line 214
   at Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer.StartIIS(Uri uri, String contentRoot) in /_/src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployer.cs:line 161
   at Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer.DeployAsync() in /_/src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployer.cs:line 113
   at Microsoft.AspNetCore.Server.IntegrationTesting.FunctionalTestsBase.DeployAsync(IISDeploymentParameters parameters) in /_/test/Common.FunctionalTests/Utilities/FunctionalTestsBase.cs:line 47
   at Microsoft.AspNetCore.Server.IIS.FunctionalTests.AppOfflineTests.DeployApp(HostingModel hostingModel) in /_/test/Common.FunctionalTests/AppOfflineTests.cs:line 229
   at Microsoft.AspNetCore.Server.IIS.FunctionalTests.AppOfflineTests.LockedAppOfflineDroppedWhileSiteIsDown_SiteReturns503(HostingModel hostingModel) in /_/test/Common.FunctionalTests/AppOfflineTests.cs:line 50
--- End of stack trace from previous location where exception was thrown ---
----- Inner Stack Trace #2 (System.DllNotFoundException) -----
   at Interop.mincore.EnumServicesStatusEx(IntPtr databaseHandle, Int32 infolevel, Int32 serviceType, Int32 serviceState, IntPtr status, Int32 size, Int32& bytesNeeded, Int32& servicesReturned, Int32& resumeHandle, String group)
   at System.ServiceProcess.ServiceController.GenerateNames()
   at System.ServiceProcess.ServiceController.GetServiceHandle(Int32 desiredAccess)
   at System.ServiceProcess.ServiceController.GenerateStatus()
   at Microsoft.Web.Administration.ServerManager.GetServiceStatus(String serviceName)
   at Microsoft.Web.Administration.Site.Stop()
   at Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer.Stop() in /_/src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployer.cs:line 367
   at Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer.StopAndDeleteAppPool() in /_/src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployer.cs:line 222
   at Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer.Dispose() in /_/src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployer.cs:line 53
   at Microsoft.AspNetCore.Server.IntegrationTesting.FunctionalTestsBase.StopServer() in /_/test/Common.FunctionalTests/Utilities/FunctionalTestsBase.cs:line 58
   at ReflectionAbstractionExtensions.DisposeTestClass(ITest test, Object testClass, IMessageBus messageBus, ExecutionTimer timer, CancellationTokenSource cancellationTokenSource) in C:\Dev\xunit\xunit\src\xunit.execution\Extensions\ReflectionAbstractionExtensions.cs:line 76
------- Stdout: -------
| [0.009s] TestLifetime Information: Starting test LockedAppOfflineDroppedWhileSiteIsDown_SiteReturns503-OutOfProcess at 2018-08-09T22:12:31
| [0.041s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: Deploying [Variation] :: ServerType=IIS, Runtime=CoreClr, Arch=x64, BaseUrlHint=, Publish=True
| [0.046s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: Executing command dotnet publish  --output "C:\BuildAgent\temp\buildTmp\cbbae145ac104990a7e5675e43218793" --framework netcoreapp2.2 --configuration Release --no-restore -p:VerifyMatchingImplicitPackageVersion=false 
| [0.351s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: dotnet-publish stdout: Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core
| [0.351s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: dotnet-publish stdout: Copyright (C) Microsoft Corporation. All rights reserved.
| [1.564s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: dotnet-publish stdout:   TestTasks -> C:\BuildAgent\work\33bdfc1cae7b2a38\modules\IISIntegration\test\TestTasks\bin\Release\netcoreapp2.2\TestTasks.dll
| [1.642s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: dotnet-publish stdout:   Microsoft.AspNetCore.Server.IISIntegration -> C:\BuildAgent\work\33bdfc1cae7b2a38\modules\IISIntegration\src\Microsoft.AspNetCore.Server.IISIntegration\bin\Release\netstandard2.0\Microsoft.AspNetCore.Server.IISIntegration.dll
| [1.830s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: dotnet-publish stdout:   OutOfProcessWebSite -> C:\BuildAgent\work\33bdfc1cae7b2a38\modules\IISIntegration\test\WebSites\OutOfProcessWebSite\bin\Release\netcoreapp2.2\OutOfProcessWebSite.dll
| [2.141s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: dotnet-publish stdout:   OutOfProcessWebSite -> C:\BuildAgent\temp\buildTmp\cbbae145ac104990a7e5675e43218793\
| [2.184s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: dotnet publish finished with exit code : 0
| [2.193s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug: Processing C:\BuildAgent\temp\buildTmp\929e64c981cc456ea097a856c908f215\runtimes\unix\lib\netcoreapp2.0
| [2.193s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying System.Security.Principal.Windows.dll
| [2.196s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug: Processing C:\BuildAgent\temp\buildTmp\929e64c981cc456ea097a856c908f215
...
```
Other tests within that build may have failed with a similar message, but they are not listed here. Check the link above for more info.

This test failed on master.

CC @muratg

This issue was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
Please use this workflow to address this flaky test issue, including checking applicable checkboxes and filling in the applicable "TODO" entries:

* Is this actually a flaky test?
  * No, this is a regular test failure, fix the test/product (TODO: Link to commit/PR)
  * Yes, proceed below...

* Is this test failure caused by product code flakiness? (Either this product, or another product this test depends on.)
  * [ ] File a bug against the product (TODO: Link to other bug)
  * Is it possible to change the test to avoid the flakiness?
    * Yes? Go to the "Change the test!" section.
    * No?
      * [ ] Disable the test (TODO: Link to PR/commit)
      * [ ] Wait for other bug to be resolved
      * [ ] Wait for us to get build that has the fix
      * [ ] Re-enable our test (TODO: Link to PR/commit)
      * [ ] Close this bug

* Is it that the test itself is flaky? This includes external transient problems (e.g. remote server problems, file system race condition, etc.)
  * Is there is a way to change our test to avoid this flakiness?
    * Yes? Change the test!
      * [ ] Change the test to avoid the flakiness, for example by using a different test strategy, or by adding retries w/ timeouts (TODO: Link to PR/commit)
      * [ ] Run the test 100 times locally as a sanity check.
      * [ ] Close this bug
    * No?
      * Is there any logging or extra information that we could add to make this more diagnosable when it happens again?
        * Yes?
            * [ ] Add the logging (TODO: Link to PR/commit)
        * No?
            * [ ] Delete the test because flaky tests are not useful (TODO: Link to PR/commit)

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
AppOfflineDroppedWhileSiteIsDown_SiteReturns503 [failed with about the same error](http://aspnetci/viewLog.html?buildId=527554&buildTypeId=XPlat_Windows_Win2008r2_Universe) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
LockedAppOfflineDroppedWhileSiteIsDown_SiteReturns503 [failed with about the same error](http://aspnetci/viewLog.html?buildId=527607&buildTypeId=XPlat_Windows_Win10Universe) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by pakrym** [comment]
Fixed in #1197.
